// @runwayml/sdk/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Uploads = exports.Workflows = exports.WorkflowInvocations = exports.Voices = exports.VoiceIsolation = exports.VoiceDubbing = exports.VideoToVideo = exports.TextToVideo = exports.TextToSpeech = exports.TextToImage = exports.Tasks = exports.SpeechToSpeech = exports.SoundEffect = exports.RealtimeSessions = exports.Organization = exports.ImageToVideo = exports.Documents = exports.CharacterPerformance = exports.Avatars = void 0;
var avatars_1 = require("./avatars.js");
Object.defineProperty(exports, "Avatars", { enumerable: true, get: function () { return avatars_1.Avatars; } });
var character_performance_1 = require("./character-performance.js");
Object.defineProperty(exports, "CharacterPerformance", { enumerable: true, get: function () { return character_performance_1.CharacterPerformance; } });
var documents_1 = require("./documents.js");
Object.defineProperty(exports, "Documents", { enumerable: true, get: function () { return documents_1.Documents; } });
var image_to_video_1 = require("./image-to-video.js");
Object.defineProperty(exports, "ImageToVideo", { enumerable: true, get: function () { return image_to_video_1.ImageToVideo; } });
var organization_1 = require("./organization.js");
Object.defineProperty(exports, "Organization", { enumerable: true, get: function () { return organization_1.Organization; } });
var realtime_sessions_1 = require("./realtime-sessions.js");
Object.defineProperty(exports, "RealtimeSessions", { enumerable: true, get: function () { return realtime_sessions_1.RealtimeSessions; } });
var sound_effect_1 = require("./sound-effect.js");
Object.defineProperty(exports, "SoundEffect", { enumerable: true, get: function () { return sound_effect_1.SoundEffect; } });
var speech_to_speech_1 = require("./speech-to-speech.js");
Object.defineProperty(exports, "SpeechToSpeech", { enumerable: true, get: function () { return speech_to_speech_1.SpeechToSpeech; } });
var tasks_1 = require("./tasks.js");
Object.defineProperty(exports, "Tasks", { enumerable: true, get: function () { return tasks_1.Tasks; } });
var text_to_image_1 = require("./text-to-image.js");
Object.defineProperty(exports, "TextToImage", { enumerable: true, get: function () { return text_to_image_1.TextToImage; } });
var text_to_speech_1 = require("./text-to-speech.js");
Object.defineProperty(exports, "TextToSpeech", { enumerable: true, get: function () { return text_to_speech_1.TextToSpeech; } });
var text_to_video_1 = require("./text-to-video.js");
Object.defineProperty(exports, "TextToVideo", { enumerable: true, get: function () { return text_to_video_1.TextToVideo; } });
var video_to_video_1 = require("./video-to-video.js");
Object.defineProperty(exports, "VideoToVideo", { enumerable: true, get: function () { return video_to_video_1.VideoToVideo; } });
var voice_dubbing_1 = require("./voice-dubbing.js");
Object.defineProperty(exports, "VoiceDubbing", { enumerable: true, get: function () { return voice_dubbing_1.VoiceDubbing; } });
var voice_isolation_1 = require("./voice-isolation.js");
Object.defineProperty(exports, "VoiceIsolation", { enumerable: true, get: function () { return voice_isolation_1.VoiceIsolation; } });
var voices_1 = require("./voices.js");
Object.defineProperty(exports, "Voices", { enumerable: true, get: function () { return voices_1.Voices; } });
var workflow_invocations_1 = require("./workflow-invocations.js");
Object.defineProperty(exports, "WorkflowInvocations", { enumerable: true, get: function () { return workflow_invocations_1.WorkflowInvocations; } });
var workflows_1 = require("./workflows.js");
Object.defineProperty(exports, "Workflows", { enumerable: true, get: function () { return workflows_1.Workflows; } });
var uploads_1 = require("./uploads.js");
Object.defineProperty(exports, "Uploads", { enumerable: true, get: function () { return uploads_1.Uploads; } });
